text/src/autogluon/text/automm/predictor.py

```diff
@@ -818,27 +971,45 @@ def _top_k_average(
             # In some cases, the training ends up too early (e.g., due to time_limit) so that there is
             # no saved best_k model checkpoints. In that scenario, we won't perform any model averaging.
             best_k_models = None
+        if best_k_models is not None:
+            if config.optimization.top_k_average_method == UNIFORM_SOUP:
+                top_k_model_paths = list(best_k_models.keys())
+            else:
+                top_k_model_paths = [
+                    v[0] for v in sorted(
+                        list(best_k_models.items()),
+                        key=lambda ele: ele[1],
+                        reverse=(minmax_mode == MAX),
+                    )
+                ]
         last_ckpt_path = os.path.join(save_path, "last.ckpt")
+        
+        return top_k_model_paths, last_ckpt_path, best_k_models_yaml_path
 
+    def _top_k_average(
+            self,
+            model,
+            save_path,
+            top_k_model_paths,
+            last_ckpt_path,
+            minmax_mode,
+            is_distill,
+            config,
```

Comment:
we can change `config` to `top_k_average_method` to be more specific.

Reply:
Good idea